test/system/irohad_test.cpp

```diff
@@ -157,6 +158,42 @@ class IrohadTest : public AcceptanceFixture {
         config_copy_, path_genesis_.string(), path_keypair_.string(), {});
   }
 
+  torii::CommandSyncClient createToriiClient(
+      bool enable_tls = false,
+      const boost::optional<uint16_t> override_port = {}) {
+    uint16_t port;
+    if (override_port) {
+      port = *override_port;
+    } else {
+      if (enable_tls) {
+        port = kSecurePort;
+      } else {
+        port = kPort;
+      }
+    }
+
+    std::unique_ptr<iroha::protocol::CommandService_v1::Stub> stub;
+    std::string stub_error;
+    if (enable_tls) {
+      iroha::network::createSecureClient<iroha::protocol::CommandService_v1>(
+          kAddress + ":" + std::to_string(port),
+          (test_data_path_ / "tls/correct.crt").string())
+          .match([&](auto secure_stub) { stub = std::move(secure_stub.value); },
+                 [&](const auto &error) { stub_error = error.error; });
+    } else {
+      stub = iroha::network::createClient<iroha::protocol::CommandService_v1>(
+          kAddress + ":" + std::to_string(port));
+    }
+
+    return torii::CommandSyncClient(
+        std::move(stub),
```

Comment:
It does not look like it will work if there was an error in `createSecureClient`.